internal/servers/destinations.go

```diff
@@ -95,16 +95,16 @@ func (s *DestinationServer) Write2(msg pb.Destination_Write2Server) error {
 				return msg.SendAndClose(&pb.Write2_Response{})
 			}
 			close(resources)
-			if err := eg.Wait(); err != nil {
-				s.Logger.Error().Err(err).Msg("got error. failed to wait for plugin")
+			if wgErr := eg.Wait(); wgErr != nil {
+				return status.Errorf(codes.Internal, "failed to receive msg: %v and wait for plugin: ", err, wgErr)
```

Comment:
I think it's not accurate to report `failed to wait for plugin`.
The error returned from the wait group is whatever returned from the wait group Go Routine, i.e. whatever returned from `s.Plugin.Write(ctx, tables, sourceName, syncTime, resources)`, so if `eg.Wait() != nil` it means the `write` failed, not that the `wait` failed.

Please let me know if that makes sense